tijanabekic/assignment1/Assignment 1.java

```diff
@@ -0,0 +1,160 @@
+Q1. Given two strings, determine if one is an anagram of the other. Two words are anagrams of
+each other if they are made of the same letters in a different order
+
+boolean isAnagram(String s1, String s2) {
+	if (s1 == null || s1 == null)
+		return false; // alternative: throw new Exception();
+	if (s1.length() != s2.length())
+		return false; // different char count
+
+	// count number of appearances of each char in s1
+	int [] charCount = new int[256]; // ASCII
+	for (int i = 0; i < s1.length(); i++) {
+		charCount[s1.charAt(i)]++;
+	}
+
+	// check if the number of appearances of each char in s2 matches
+	for (int i = 0; i < s2.length(); i++) {
+		if (charCount[s2.charAt(i)] <= 0)
+			return false; // char s2.charAt(i) appears more times in s2
+		charCount[s2.charAt(i)]--;
+	}
+	return true;
+}
+
+Follow-ups:
+1. Make the algorithm able to handle both case sensitive and case insensitive anagrams.
+
+// case insensitive
+boolean isAnagramCaseInsensitive(String s1, String s2) {
+	if (s1 == null || s1 == null)
+		return false; // alternative: throw new Exception();
+	if (s1.length() != s2.length())
+		return false; // different char count
+
+	s1.toLowerCase();
+	s2.toLowerCase();
+	// count number of appearances of each char in s1
+	int [] charCount = new int[256]; // ASCII
+	for (int i = 0; i < s1.length(); i++) {
+		charCount[s1.charAt(i)]++;
+	}
+
+	// check if the number of appearances of each char in s2 matches
+	for (int i = 0; i < s2.length(); i++) {
+		if (charCount[s2.charAt(i)] <= 0)
+			return false; // char s2.charAt(i) appears more times in s2
+		charCount[s2.charAt(i)]--;
+	}
+	return true;
+}
+2. Make the algorithm able to handle anagrams of sentences, where each word in the resulting
+sentence is an anagram of one of the words in the original sentence
+
+class WordInfo {
+	private int count = 0;
+	private int length = 0;
+	public WordInfo(int cnt, int len) {
+		count = cnt;
+		length = len;
+	}
+	public void setCount(int cnt) {
+		count = cnt;
+	}
+	public int getCount() {
+		return count;
+	}
+	public int getLength() {
+		return length;
+	}
+}
+
+boolean isAnagramSentence(String s1, String s2) {
+	if (s1 == null || s1 == null)
+		return false; // alternative: throw new Exception();
+	if (s1.length() != s2.length())
+		return false; // different char count
+
+	// case insensitive
+	s1.toLowerCase();
+	s2.toLowerCase();
+
+
+	HashMap<String, WordInfo> map = new HashMap<String, WordInfo>();
+	StringBuilder currWord = new StringBuilder();
+
+	// find all words in the first sentence
+	for (int i = 0; i < s1.length(); i++) {

```

Comment:
This code could be made simpler by using the String method split(" "). It takes a string and returns an array of strings split by the given character. So "foo bar" would return ["foo", "bar"]